src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/EditAppealAfterSubmitHandler.java

```diff
@@ -65,14 +66,26 @@ public boolean canHandle(
             callback
                 .getCaseDetails()
                 .getCaseData();
+        LocalDate decisionDate;
 
-        Optional<String> maybeHomeOfficeDecisionDate = asylumCase.read(HOME_OFFICE_DECISION_DATE);
+        if (asylumCase.read(OUT_OF_COUNTRY_DECISION_TYPE, OutOfCountryDecisionType.class).map(
+            value -> OutOfCountryDecisionType.REMOVAL_OF_CLIENT.equals(value)).orElse(false)) {
+            Optional<String> maybeHomeOfficeDecisionLetterDate = asylumCase.read(DECISION_LETTER_RECEIVED_DATE);
 
-        LocalDate homeOfficeDecisionDate =
-            parse(maybeHomeOfficeDecisionDate
-                .orElseThrow(() -> new RequiredFieldMissingException("homeOfficeDecisionDate is missing")));
+            decisionDate = parse(maybeHomeOfficeDecisionLetterDate
+                .orElseThrow(() -> new RequiredFieldMissingException("decisionLetterDateReceived is missing")));
 
-        if (homeOfficeDecisionDate.isBefore(dateProvider.now().minusDays(appealOutOfTimeDays))) {
+        } else {
+
+
+            Optional<String> maybeHomeOfficeDecisionDate = asylumCase.read(HOME_OFFICE_DECISION_DATE);
+
+            decisionDate =
+                parse(maybeHomeOfficeDecisionDate
+                    .orElseThrow(() -> new RequiredFieldMissingException("homeOfficeDecisionDate is missing")));
+        }
+
+        if (decisionDate.isBefore(dateProvider.now().minusDays(appealOutOfTimeDays))) {
```

Comment:
Could you please find out where SUBMISSION_OUT_OF_TIME is used ?
I am just curious to know why we need this handler.